nativepython/type_wrappers/python_object_of_type_wrapper.py

```diff
@@ -91,18 +93,36 @@ def convert_to_type_with_target(self, context, e, targetVal, explicit):
         if not explicit:
             return super().convert_to_type_with_target(context, e, targetVal, explicit)
 
-        if target_type.typeRepresentation == Int64:
+        arith_calls = {
+            Int64: runtime_functions.pyobj_to_int64,
+            Int32: runtime_functions.pyobj_to_int32,
+            Int16: runtime_functions.pyobj_to_int16,
+            Int8: runtime_functions.pyobj_to_int8,
+            UInt64: runtime_functions.pyobj_to_uint64,
+            UInt32: runtime_functions.pyobj_to_uint32,
+            UInt16: runtime_functions.pyobj_to_uint16,
+            UInt8: runtime_functions.pyobj_to_uint8,
+            Float64: runtime_functions.pyobj_to_float64,
+            Float32: runtime_functions.pyobj_to_float32
+        }
+
+        t = target_type.typeRepresentation
+        if t in arith_calls:
             context.pushEffect(
                 targetVal.expr.store(
-                    runtime_functions.pyobj_to_int.call(e.nonref_expr)
+                    arith_calls[t].call(e.nonref_expr)
                 )
             )
             return context.constant(True)
 
-        if target_type.typeRepresentation == UInt64:
+        tp = getTypePointer(t)
```

Comment:
please make this a method on ExpressionConversionContext, so that we will end up having exactly one place where this function gets called.